modules/misc/qt.nix

```diff
@@ -147,37 +182,31 @@ in {
       '';
     }];
 
-    qt.style.package = mkIf (cfg.style.name != null)
-      (mkDefault (stylePackages.${toLower cfg.style.name} or null));
-
-    home.sessionVariables = envVars;
+    qt.style.package = lib.mkIf (cfg.style.name != null)
+      (lib.mkDefault (stylePackages.${lib.toLower cfg.style.name} or null));
+
+    home = {
+      sessionVariables = envVars;
+      # home.sessionVariables does not support setting the same environment
+      # variable to different values.
+      # Since some other modules may set the QT_PLUGIN_PATH or QML2_IMPORT_PATH
+      # to their own value, e.g.: fcitx5, we avoid conflicts by setting
+      # the values in home.sessionVariablesExtra instead.
+      sessionVariablesExtra = ''
+        export QT_PLUGIN_PATH=${envVarsExtra.QT_PLUGIN_PATH}
+        export QML2_IMPORT_PATH=${envVarsExtra.QML2_IMPORT_PATH}
+      '';
```

Comment:
I know that using `home.sessionVariables` here is not ideal, but this is the only solution I found for now.

We probably need something similar to `environment.profileRelativeEnvVars` or `environment.profileRelativeSessionVariables` in NixOS to fix this issue once for all.

Reply:
This is at least better than the previous solution I did: https://github.com/nix-community/home-manager/pull/4579/commits/3452e14ec729e0582a5fb120c14ed5c6d7fb64d3

Reply:
> We probably need something similar to environment.profileRelativeEnvVars or environment.profileRelativeSessionVariables in NixOS to fix this issue once for all.

Or even simpler, as long as we have something that allows a similar semantic as `home.sessionPath`, but generic (e.g.: `home.searchPathVariables.QT_PLUGIN_PATH = [ ... ]`), this would already work.

Reply:
I decided to fix this issue in a separate PR: https://github.com/nix-community/home-manager/pull/4582.

I think this PR can still be merged before that PR above. In any case once the other PR is accepted I will change this PR to re-implement this in terms of `home.sessionSearchVariables`.